nodes/nomos-node-api/src/http/info.rs

```diff
@@ -0,0 +1,71 @@
+use std::{fmt::Debug, hash::Hash};
+
+use consensus_engine::overlay::{RandomBeaconState, RoundRobin, TreeOverlay};
+use full_replication::Certificate;
+use nomos_consensus::{
+    network::adapters::libp2p::Libp2pAdapter as ConsensusLibp2pAdapter, CarnotConsensus,
+    CarnotInfo, ConsensusMsg,
+};
+use nomos_core::{
+    da::{
+        blob,
+        certificate::{self, select::FillSize as FillSizeWithBlobsCertificate},
+    },
+    tx::{select::FillSize as FillSizeWithTx, Transaction},
+};
+use nomos_mempool::{
+    backend::mockpool::MockPool, network::adapters::libp2p::Libp2pAdapter as MempoolLibp2pAdapter,
+};
+use nomos_network::{
+    backends::libp2p::{Command, Libp2p, Libp2pInfo},
+    NetworkMsg, NetworkService,
+};
+use nomos_storage::backends::{sled::SledBackend, StorageSerde};
+use serde::{de::DeserializeOwned, Serialize};
+use tokio::sync::oneshot;
+
+pub type Carnot<Tx, SS, const SIZE: usize> = CarnotConsensus<
```

Comment:
I think we will have a cyclic dependencies problem if I add `nomos-node` as a dependency to `nomos-node-api`. Because in the integrated PR #490, the `nomos-node` also depends on `nomos-node-api` crate. But if we can move the `nomos-node` binary to a new crate, we can avoid the cyclic dependency problem.

Reply:
Yes, I think this is a signal that nomos-node need to have some shared lib.

Reply:
What about `nomos-node-types`? A shared lib for types will also help us have fewer generics in the HTTP routes which is helpful for issue #491.

Reply:
nomos-node already has a lib, we can use that

Reply:
If we do not import this into the lib should be ok then yes.

Reply:
I do not think we can use `nomos-node`, even though it already a lib. We have cycle dependency problem.

<img width="1362" alt="image" src="https://github.com/logos-co/nomos-node/assets/42351146/e727ab35-5c06-4fea-8e72-93da6a28f40b">

Reply:
I think the only usage of nomos-node-api inside of nomos-node is in the definition of the `Nomos` services struct. What if we move that in the bin and leave the type definitions + config in the lib? I don't think the full services struct is ever used

Reply:
Yeah, the only usage is `Nomos`, if we move it to the bin, when we are good.